ui/src/app/groupings/[groupingPath]/_components/export-dropdown.tsx

```diff
@@ -1,17 +1,107 @@
+import { GroupingMembers } from '@/lib/types';
+import { Spinner } from '@/components/ui/spinner';
+
+const ExportDropdown = ({ groupingPath }: { groupingPath: string }) => {
+    const groups = [
+        { type: '', label: 'All Members', icon: faUsers },
+        { type: 'basis', label: 'Basis', icon: faIdCard },
+        { type: 'include', label: 'Include', icon: faUserPlus },
+        { type: 'exclude', label: 'Exclude', icon: faUserMinus }
+    ];
+
+    const [isPending, startTransition] = useTransition();
+
+    const exportCsv = async (group: string) => {
+        startTransition(async () => {
+            const groupPath = group === '' ? groupingPath : `${groupingPath}:${group}`;
+            const sortString = 'name';
+            const isAscending = true;
+            const data: GroupingMembers = await getGroupingMembers(groupPath, sortString, isAscending);
+
+            const members = data?.members;
+            const groupName = groupingPath.split(':').pop();
+            const headers = ['Last', 'First', 'Username', 'UH Number', 'Email'];
+
+            const rows = members.map((member) => {
+                let email = '';
+                if (member.uid && /^[a-zA-Z0-9._%+-]+$/.test(member.uid)) {
+                    email = `${member.uid}@hawaii.edu`;
+                }
+                return [member.lastName, member.firstName, member.uid, member.uhUuid, email];
+            });
+
+            const csvContent = [headers, ...rows].map((row) => row.join(',')).join('\n');
+
+            const blob = new Blob([csvContent], { type: 'text/csv' });
+            const url = URL.createObjectURL(blob);
+            const link = document.createElement('a');
+            link.href = url;
+            link.download = group === '' ? `${groupName}_members_list.csv` : `${groupName}_${group}_list.csv`;
+            link.click();
+            URL.revokeObjectURL(url);
+        });
+    };
 
-const ExportDropdown = () => {
     return (
-        <div className="btn-group dropdown float-right">
-            <Button aria-label="Export Grouping">
-                <FileInput className="mr-1" />
-                Export Grouping
-                <ChevronDown className="ml-1" />
-            </Button>
-        </div>
+        <form className="btn-group float-right relative">
+            <TooltipProvider>
+                <Tooltip>
+                    <TooltipTrigger asChild>
+                        <DropdownMenu>
+                            <DropdownMenuTrigger asChild>
+                                <Button
+                                    aria-label="Export Grouping"
+                                    variant="light"
+                                    className="px-3 py-1.5 text-base"
+                                    disabled={isPending}
+                                >
+                                    {isPending ? (
+                                        <Spinner size="sm" />
```

Comment:
Add `className="mr-1"` to the Spinner to match the position of the faFileExport icon.